host-components/src/main.rs

```diff
@@ -0,0 +1,393 @@
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate serde_derive;
+
+use ::host_components::*;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use warp::Filter;
+
+#[tokio::main]
+async fn main() {
+    let socket = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), BIND_PORT);
+
+    let keeploaderlist = models::find_existing_keep_loaders();
+
+    // TODO: in case there are existing keeploaders out there, try finding them at
+    //  /tmp/enarx-keep-*.sock, try connecting to them, if OK, add to
+    //  keeploaderlist
+    // TODO: what if we can't connect?  should we delete them?  Mark them dead?
+    // Match any non-explicitly managed requests "/" request and return
+    let declare = warp::any().map(|| {
+        format!(
+            "Protocol_name = {}\nProtocol_version = {}",
+            PROTO_NAME, PROTO_VERSION
+        )
+    });
+
+    let keep_posts = warp::post()
+        .and(warp::path("keeps_post"))
+        .and(warp::body::json())
+        .and(filters::with_keeploaderlist(keeploaderlist.await))
+        .and_then(filters::keeps_parse);
+
+    let routes = keep_posts.or(declare);
+    println!(
+        "Starting server on {}, {} v{}",
+        BIND_PORT, PROTO_NAME, PROTO_VERSION
+    );
+    warp::serve(routes)
+        .tls()
+        .cert_path("key-material/server.crt")
+        .key_path("key-material/server.key")
+        .run(socket)
+        .await;
+}
+
+mod models {
+    use ::host_components::*;
+    use glob::glob;
+    use std::io::prelude::*;
+    use std::os::unix::net::UnixStream;
+    use std::sync::Arc;
+    use tokio::sync::Mutex;
+
+    pub fn new_empty_keeploaderlist() -> KeepLoaderList {
+        Arc::new(Mutex::new(Vec::new()))
+    }
+
+    pub async fn find_existing_keep_loaders() -> KeepLoaderList {
+        println!("Looking for existing keep-loaders in /tmp");
```

Comment:
no `/tmp` please

Reply:
What's wrong with the keep-loader 109-112 example?